### PR TITLE
Add unit testing to update

### PR DESCRIPTION
**Pull request by JosephSalisbury**
Fixes https://github.com/giantswarm/inago/issues/131
- [x] Add tests for methods in `update.go`
- [x] Fix `go vet` error: `controller/update_test.go:99: arg test.errMatcher in printf call is a function value, not a function call`

### CODE REVIEW

**Comment by JosephSalisbury** [comment]
@giantswarm/cena RFR - I could spend ages on this, but I've got things to do, so I'm going to stop here.

@zeisss I couldn't hack that error issue, so I've removed the offending test. We'll see if it happens again.

Further, I've logged https://github.com/giantswarm/inago/issues/171 to further update testing.

**Comment by Nesurion** [comment]
can we get more comments for the tests? I was thinking one short comment before each test case, explaining what the test case.

**Comment by Nesurion** [comment]
LGTM

File: controller/controller_test.go [hunk 1]
```diff
@@ -854,3 +854,36 @@ func TestController_WaitForStatus_Timeout(t *testing.T) {
...
+				Name: "test-main@1.service",
+			},
+		},
+		nil,
+	)
+
+	request := Request{
+		RequestConfig: RequestConfig{
+			Group:    "test",
+			SliceIDs: []string{"1"},
```
Comment (zeisss) [inline controller/controller_test.go:875]: Do SliceIDs make sense for an Update? I would expect their presence to raise an error.

File: controller/update_test.go [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+	var tests = []struct {
+		function   func(context.Context, Request) (*task.Task, error)
+		ctx        context.Context
+		req        Request
+		errMatcher func(err error) bool
+	}{
+		// Test a task that does nothing.
+		{
+			function: func(ctx context.Context, req Request) (*task.Task, error) {
+				taskObject, _ := testController.TaskService.Create(
```
Comment (zeisss) [inline controller/update_test.go:55]: why not just `return testController...`?

**Comment by xh3b4sd** [comment]
Here are open questions. Do we need to address them?

**Comment by zeisss** [comment]
The first one might be nice to be discussed.

**Comment by JosephSalisbury** [comment]
Bah, I seem to have missed these. Uncertain.

**Pull request merged**